src/player.h

```diff
@@ -1152,6 +1155,8 @@ class player : public Character, public JsonSerializer, public JsonDeserializer
         int cached_turn;
         tripoint cached_position;
 
+        int fell_asleep_turn;
```

Comment:
And it's a bit hacky, why don't you add the start time to the effect data? This would guarantee that it's properly set up (in `Creature::add_effect`) and it might be useful for other effects as well.

Oh, and very nitpicky and theoretically: the value is read when a player object is copied as the compiler generated copy-constructor / assignment operator reads it. And this happens for example when a game is loaded and `g->u` is reset with `u = player();` in `game::load`.

But other members of the player class (or other classes) might already be used uninitialized, it wouldn't surprise me. And it's a highly theoretically problem anyway.